File: manga_fe/src/class/api/ResponseAPI.ts
class ResponseAPI {
  private readonly _message: string;
  private readonly _statusCode: number;

  constructor(message: string, statusCode: number) {
    this._message = message;
    this._statusCode = statusCode;
  }

  get message(): string {
    return this._message
  }

  get statusCode(): number {
    return this._statusCode
  }
}

export default ResponseAPI;

File: manga_be/src/main/java/com/reader/manga/adapters/input/dtos/ResponseAPI.java
package com.reader.manga.adapters.input.dtos;

import lombok.Builder;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@Builder
public record ResponseAPI(
        @NotNull @NotBlank String message,
        @NotNull Integer statusCode,
        Object responseObject
) {}